app/models/embed/purl/resource_file.rb

```diff
@@ -40,15 +52,34 @@ def stacks_url
       end
 
       def label_or_filename
+        return caption_label if caption?
+
         label.presence || filename
       end
 
+      def language_label
+        FILE_LANGUAGE_CAPTION_LABELS[language_code]
```

Comment:
should there be a default here, in case we find a lang we don't know about?

Reply:
In the earlier approach, if a language code exists but is not recognized in the mapping, the label returned is "Caption"

Reply:
"Caption" is also returned by the media tag component in this case

Reply:
The other behavior in the former approach was that if there was no language code, "en" would be returned. I see that in the language_code method on line 73 below so that seems fine.

Reply:
@jcoyne Yes, because `language_code` defaults to `en`, the effective default for `language_label` is `English`. I believe Dinah has said before that English as the default caption language is preferred.

Reply:
@jcoyne the last couple times we talked about it, Dinah opted for "Caption file" when the language is unknown, IIRC. I can patch in that change.

Reply:
@jcoyne Made the change. Ready for another look.